### ISSUE TITLE
Netflix doesnt play

### ISSUE THREAD

**Issue by poiromaniax** [slashlos/Helium#25]
Hey,

Thanks for picking up this project.

Sadly, netflix wont play for me - just a "something went wrong" error. Same as Helium 2

**Comment by slashlos** [comment]
Reproducer please

**Comment by slashlos** [comment]
With updated user agent - see [issue #22](https://github.com/slashlos/Helium/issues/22), Netflix seems ok in my limited testing.

**Comment by poiromaniax** [comment]
> With updated user agent - see [issue #22](https://github.com/slashlos/Helium/issues/22), Netflix seems ok in my limited testing.

I tried adding chrome string to user agent and now get this loop

![image](https://user-images.githubusercontent.com/15910083/62443115-b4d44900-b762-11e9-84ae-823093b04506.png)

**Comment by poiromaniax** [comment]
In fact, this is what I now get even when falling back to the default user agent

**Comment by slashlos** [comment]
Hmmm not getting such; need steps to reproduce, version info and user agent string, thanks.

**Comment by poiromaniax** [comment]
Open Helium, Enter URL (netflix.com)
Select show, click play. Error occurs:

> There was an unexpected error. Please reload the page and try again.
> Error Code: S7363-1260-FFFFD089

Helium version 3.0.4162 (latest from Releases page)
User Agent (default) : Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/601.6.17 (KHTML, like Gecko) Version/9.1.1 Safari/601.6.17

If I change the agent string to `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/601.6.17 (KHTML, like Gecko) Version/9.1.1 Chrome/75.0.3770.142 Safari/601.6.17`

Then I get the "please reload the page" error. Even if I go and choose defualt user agent. I have to quit helium and reopen it but then I get the "unexpected error" all over again

**Comment by slashlos** [comment]
Thanks; typically if you're shown an error code, you can follow their help screens to find out why.  

In your case, the error seems to refer to the type of monitor used - requires HDCP.  

In my testing, adding the chrome bits (`Chrome/75.0.3770.142`) isn't a panacea and sometimes made things worse.  I'm watching Stranger Things without it fine.

**Comment by slashlos** [comment]
Note:  the tail end of the UA string above should be `Version/9.1.1 Safari/601.6.17 Chrome/75.0.3770.142` - the `Version/9.1.1` applies to `Safari/601.1.17`, and the ordering - of Safari and Chrome entries I do not believe matter?

**Comment by poiromaniax** [comment]
> Thanks; typically if you're shown an error code, you can follow their help screens to find out why.
> 
> In your case, the error seems to refer to the type of monitor used - requires HDCP.
> 
> In my testing, adding the chrome bits (`Chrome/75.0.3770.142`) isn't a panacea and sometimes made things worse. I'm watching Stranger Things with or without it.
> 
> I'm thinking of adding a simpler interface to altering this.

Yes I checked their documentation but its not relevant. Netflix works in any other browser which Ive tried (Safari and Chrome) - just does not work in Helium. Maybe some sort of metadata isnt being passed through by helium?

**Comment by slashlos** [comment]
I suspect the user-agent string.  I'm not a fan of altering it but rather than the historical precedent of having it set by the app, I'm deciding to capture it from the app kit framework.  The user can however alter a global preference to be inherited by views as well as to set a per-view custom user agent; this setting will be restored on a per-url basis.  I use NetFlix fine with the default (framework) user agent; currently watching Stranger Things; btw you might want to enable location services which might come into play here?  HTH; but as to Hulu, I'm still trying to sort out :-(

**Comment by poiromaniax** [comment]
Ive reset the UA back to default, and enabled location services but still get Error Code: S7363-1260-FFFFD089 on playback.

I have no idea what else to actually do

As I say, netflix works just peachy in other browsers - its only helium

**Comment by slashlos** [comment]
Their error seems to point a finger to your env. of h/w, and/or browser setup - i.e., UA. is my bet.

Try as a UA, the framework kit default string below - copy-n-paste it.  Note that the o/s version might be different for you:  also, is it all series or just a particular one; I'd like to test along?  

`Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_5) AppleWebKit/605.1.15 (KHTML, like Gecko)`

In my case I'm testing with rMBP 13" with twin A1407 apple monitors (10.14.6) , 2012 MINI a ROKU (TCL) TV (10.14.5), and a rMPP 15", all just peachy; even share from Safari as Apple requires just dandy.  

I test across osx releases I have on hand (.13 & .14), so I'm curious for details re: your environment as I've had to leave .12 due to app kit requirements - are you running .12 ?

**Comment by poiromaniax** [comment]
Using this new UA I get a different error page S7381-1204 - Error with HTML5 player

Its anything that I attempt to play - nothing specific

Im using a hackintosh (although I dont think that matters if it plays in a native browser), twin Dell U2313HM Monitors

Using macOS 10.14.5 currently

**Comment by poiromaniax** [comment]
I saw a thread on Reddit now that you cant use helium with netflix and dual monitors? Something about DRM

**Comment by slashlos** [comment]
If it’s hd content it might matter your h/w; the native UA and all apple h/w runs for me.  It might be that the bits in the web app kit just aren’t up to be a replacement yet.  A guess, but try a different monitor?

**Comment by slashlos** [comment]
p.s., testing build 4162 on 10.14.6 - 13" rMBP, dual Apple 27" - *no problems*; what I did ...

1st monitor, open URL from dock menu: www.netflix.com; play The Vietnam War series; drag to 2nd monitor, and
2nd monitor open URL from dock menu: www.netflix.com; play Dark series; drag tp 1st monitor.

So, played 2 different series content on 2 different monitors, and back on 1st monitor, repeat sequence to play Stranger Things on 1st monitor; no problems.  Try this sequence on your h/w might indicate a platform issue ?
